test-packages/test-utils/src/utils/download-ui5-resources.ts

```diff
@@ -0,0 +1,115 @@
+// This file can be used to download api.json files for UI5 version we want to add to the test directory.
+// It is run from the download-resources script in the package.json.
+// To get a different version change the version number in the last line of this file.
+import { TestModelVersion } from "../../api";
+import { reduce, forEach, keys } from "lodash";
+import { dirname, resolve } from "path";
+import { writeFileSync, mkdirsSync, readFileSync } from "fs-extra";
+import fetch from "node-fetch";
+
+async function getLibs(): Promise<string[]> {
+  // To fetch the metadata.json file dynamically by version do the following:
```

Comment:
I don't understand "coding in comments"
If you want to support a dynamic flow, add such support as an optional flow directly why the complexity?

Reply:
Mostly because I didn't want to change the `tsconfig.base.json` to support dom APIs for a test utility script, and I couldn't find a way to compile it when changing only the `tsconfig.json` of the test package. It's only required for the `jszip` usage but I couldn't find another library with nice API for reading zip files (I didn't spend a lot of time on it though).

Reply:
If we keep the api.json files as part of the source code, then perhaps this code
should not be part of the test-utilities/src and instead be located as part of test-utilities/scripts and then you can **write it in JavaScript** and avoid the TypeScript overhead.

Even if we do want to use it in automated test flows (download api.json on demand in tests) rather then manual execution. We can still put it in a **different** package under test-packages folder, and implement said packages purely in JavaScript.

btw we seem to be throwing everything into a single monolithic test-utilities package
which already created a cyclic package dependency... (unrelated to this PR)

Reply:
There are no comments now (also this is no longer a standalone script)